HousingManagementSystemApi.Tests/GatewaysTests/AssetGatewayTests.cs

```diff
@@ -0,0 +1,107 @@
+
+
+        public AssetGatewayTests()
+        {
+            mockHttp = new MockHttpMessageHandler();
+
+            mockHttp.When($"*assets/assetId/{AssetId}")
+                .Respond("application/json", assetResponse);
+
+            var httpClient = mockHttp.ToHttpClient();
+            httpClient.BaseAddress = new Uri("http://localhost/");
+
+            var httpClientFactory = new Mock<IHttpClientFactory>();
+            httpClientFactory.Setup(x => x.CreateClient(It.IsAny<string>())).Returns(httpClient);
+            systemUnderTest = new AssetGateway(httpClientFactory.Object);
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidArgumentTestData))]
+#pragma warning disable xUnit1026
+#pragma warning disable CA1707
+        public async void GivenInvalidAssetIdArgument_WhenRetrievingAnAsset_ThenAnExceptionIsThrown<T>(T exception, string assetId) where T : Exception
+#pragma warning restore CA1707
+#pragma warning restore xUnit1026
+        {
+            // Arrange
+
+            // Act
+            Func<Task> act = async () => await systemUnderTest.RetrieveAsset(assetId);
+
+            // Assert
+            await act.Should().ThrowExactlyAsync<T>();
+        }
+
+        public static IEnumerable<object[]> InvalidArgumentTestData()
+        {
+            yield return new object[] { new ArgumentNullException(), null };
+            yield return new object[] { new ArgumentException(), "" };
+            yield return new object[] { new ArgumentException(), " " };
+        }
+
+        [Fact]
+#pragma warning disable CA1707
+        public async void GivenAAssetIdArgument_WhenRetrievingAnAsset_ThenNoExceptionIsThrown()
+#pragma warning restore CA1707
+        {
+            // Arrange
+
+            // Act
+            Func<Task> act = async () => await systemUnderTest.RetrieveAsset(AssetId);
+
+            // Assert
+            await act.Should().NotThrowAsync();
+        }
+
+        [Fact]
+#pragma warning disable CA1707
+        public async void GivenAAssetId_WhenRetrievingAnAsset_ThenAddressesAreRetrievedFromApi()
```

Comment:
Please rename to be grammatically correct (A -> An), i.e.
```suggestion
        public async void GivenAnAssetId_WhenRetrievingAnAsset_ThenAddressesAreRetrievedFromApi()
```